backend/app/api/endpoints/internal/chat_storage.py

```
@@ -385,6 +426,162 @@ def _build_user_message_content(
+        logger.debug(
+            f"[_fetch_kb_documents_content] Loaded {len(content_parts)} documents "
+            f"for kb_id={knowledge_id}, total_chars={sum(len(p) for p in content_parts)}"
+        )
+
+    return "\n\n".join(content_parts)
+
+
+def _fetch_kb_head_documents_content(
+    db: Session,
+    document_ids: list[int],
+    offset: int = 0,
```

Ruff (0.14.14)

[warning] Unused function argument: `offset`

(ARG001)